qa/smoke-test-http/src/javaRestTest/java/org/elasticsearch/http/BlockedSearcherRestCancellationTestCase.java

```diff
@@ -149,12 +155,36 @@ private static class SearcherBlockingEngine extends ReadOnlyEngine {
 
         @Override
         public Searcher acquireSearcher(String source, SearcherScope scope, Function<Searcher, Searcher> wrapper) throws EngineException {
+            if (testCaseLogger.isDebugEnabled()) {
+                testCaseLogger.debug(
+                    Strings.format(
+                        "in acquireSearcher for shard [%s] on thread [%s], availablePermits=%d",
+                        config().getShardId(),
+                        Thread.currentThread().getName(),
+                        searcherBlock.availablePermits()
+                    ),
+                    new ElasticsearchException("stack trace")
```

Comment:
IIUC, this throws an exception every time debug() is called??

Reply:
No, it just constructs the exception so we have a stack trace to log.